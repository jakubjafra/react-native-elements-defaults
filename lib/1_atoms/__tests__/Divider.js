import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

const utils = require("../../utils");

describe('with no theme applied', () => {
    let OldDivider, Divider;

    beforeAll(() => {
        let defaultStyles = {
            'atoms/divider': {
            }
        };

        OldDivider = require('react-native-elements/src/divider/Divider').default;

        Divider = utils.connectTheme(true, 'atoms/divider', require('../Divider'))(defaultStyles);
    });

    test('OldDivider and Divider should be different', () => {
        expect(OldDivider).not.toBe(Divider);
    });

    test('OldDivider renders correctly', () => {
        const tree = renderer.create(<OldDivider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Divider renders correctly', () => {
        const tree = renderer.create(<Divider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldDivider and Divider returns the same thing if no theme is applied', () => {
        const oldTree = renderer.create(<OldDivider />).toJSON();
        const tree = renderer.create(<Divider />).toJSON();

        expect(JSON.stringify(oldTree)).toEqual(JSON.stringify(tree));
    });
});

describe('with basic theme applied', () => {
    let OldDivider, Divider;

    beforeAll(() => {
        let defaultStyles = {
            'atoms/divider': {
                style: {
                    backgroundColor: 'red',
                }
            }
        };

        OldDivider = require('react-native-elements/src/divider/Divider').default;

        Divider = utils.connectTheme(true, 'atoms/divider', require('../Divider'))(defaultStyles);
    });

    test('OldDivider and Divider should be different', () => {
        expect(OldDivider).not.toBe(Divider);
    });

    test('OldDivider renders correctly', () => {
        const tree = renderer.create(<OldDivider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Divider renders correctly', () => {
        const tree = renderer.create(<Divider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldDivider and Divider should be different caused by theme applied', () => {
        const oldTree = renderer.create(<OldDivider />).toJSON();
        const tree = renderer.create(<Divider />).toJSON();

        expect(oldTree).not.toEqual(tree);
    });
});
