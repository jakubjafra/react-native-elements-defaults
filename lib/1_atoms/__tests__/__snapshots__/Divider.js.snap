// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with basic theme applied Divider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;

exports[`with basic theme applied OldDivider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`with no theme applied Divider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`with no theme applied OldDivider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;
